packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -350,7 +350,16 @@ export class MirrorNodeClient {
     handleError(error: any, path: string, pathLabel: string, effectiveStatusCode: number, method: REQUEST_METHODS, requestIdPrefix?: string) {
         const mirrorError = new MirrorNodeClientError(error, effectiveStatusCode);
         const acceptedErrorResponses = MirrorNodeClient.acceptedErrorStatusesResponsePerRequestPathMap.get(pathLabel);
-        if (error.response && acceptedErrorResponses && acceptedErrorResponses.indexOf(effectiveStatusCode) !== -1) {
+
+        const acceptError = (
+            // Handles a FAIL_INVALID error thrown in the case of a non-existing contract in /contracts/call
```

Comment:
This should be managed at the `eth_x` method level

Reply:
`handleError` should really only manage fundamental retries and then throw a valid `MirrorNOdeClientError`.
Eth methods should decide for themselves how to handle the errors.